### ISSUE TITLE
Can't resolve build. Any clues on what I'm doing wrong?

### ISSUE THREAD

**Issue by jcklpe** [joezimjs/Webpack-Introduction-Tutorial#4]
Hey Joe, 

Excellent article so far. But I'm also running into some problems.. 

I just finished adding the 

…
  "scripts": {
    "build": "webpack src/main.js dist/bundle.js",
  }
…

To the package.json and when I run `npm run build` it gives me the following error:

```

> webpack-practice@1.0.0 build /mnt/c/Users/Owner/Work/Web/Webpack-Practice
> webpack src/main.js dist/bundle.js

Hash: 72244a08ceb38dfe1e1e
Version: webpack 4.12.0
Time: 425ms
Built at: 06/18/2018 4:34:23 PM
 1 asset
 [25] (webpack)/buildin/module.js 497 bytes {0} [built]
[107] (webpack)/buildin/global.js 489 bytes {0} [built]
[123] ./src/main.js 122 bytes {0} [built]
[124] multi ./src/main.js dist/bundle.js 40 bytes {0} [built]
    + 121 hidden modules

WARNING in configuration
The 'mode' option has not been set, webpack will fallback to 'production' for this value. Set 'mode' option to 'development' or 'production' to enable defaults for each environment.
You can also set it to 'none' to disable any default behavior. Learn more: https://webpack.js.org/concepts/mode/

ERROR in multi ./src/main.js dist/bundle.js
Module not found: Error: Can't resolve 'dist/bundle.js' in '/mnt/c/Users/Owner/Work/Web/Webpack-Practice'
 @ multi ./src/main.js dist/bundle.js

npm ERR! Linux 4.4.0-17134-Microsoft
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "build"
npm ERR! node v8.10.0
npm ERR! npm  v3.5.2
npm ERR! code ELIFECYCLE
npm ERR! webpack-practice@1.0.0 build: `webpack src/main.js dist/bundle.js`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the webpack-practice@1.0.0 build script 'webpack src/main.js dist/bundle.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the webpack-practice package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     webpack src/main.js dist/bundle.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs webpack-practice
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls webpack-practice
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /mnt/c/Users/Owner/Work/Web/Webpack-Practice/npm-debug.log
```

Also here's the node debug log:

```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'run', 'build' ]
2 info using npm@3.5.2
3 info using node@v8.10.0
4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
5 info lifecycle webpack-practice@1.0.0~prebuild: webpack-practice@1.0.0
6 silly lifecycle webpack-practice@1.0.0~prebuild: no script for prebuild, continuing
7 info lifecycle webpack-practice@1.0.0~build: webpack-practice@1.0.0
8 verbose lifecycle webpack-practice@1.0.0~build: unsafe-perm in lifecycle true
9 verbose lifecycle webpack-practice@1.0.0~build: PATH: /usr/share/npm/bin/node-gyp-bin:/mnt/c/Users/Owner/Work/Web/Webpack-Practice/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/mnt/c/Program Files/Oculus/Support/oculus-runtime:/mnt/c/Program Files (x86)/Razer Chroma SDK/bin:/mnt/c/Program Files/Razer Chroma SDK/bin:/mnt/c/Program Files (x86)/Common Files/Intel/Shared Libraries/redist/intel64_win/compiler:/mnt/c/Windows/System32:/mnt/c/Windows:/mnt/c/Windows/System32/wbem:/mnt/c/Windows/System32/WindowsPowerShell/v1.0:/mnt/c/Windows/System32/OpenSSH:/mnt/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/mnt/c/Program Files/Common Files/Autodesk Shared:/mnt/c/Program Files (x86)/Autodesk/Backburner:/mnt/c/Program Files/Perforce:/mnt/c/ProgramData/chocolatey/bin:/mnt/c/Program Files/Git/cmd:/mnt/c/Program Files/Git LFS:/mnt/c/Program Files/dotnet:/mnt/c/Program Files/Microsoft SQL Server/130/Tools/Binn:/mnt/c/Users/Owner/AppData/Local/Microsoft/WindowsApps:/mnt/c/Users/Owner/AppData/Local/hyper/app-2.0.0/resources/bin:/mnt/c/Program Files/Microsoft VS Code/bin:/snap/bin
10 verbose lifecycle webpack-practice@1.0.0~build: CWD: /mnt/c/Users/Owner/Work/Web/Webpack-Practice
11 silly lifecycle webpack-practice@1.0.0~build: Args: [ '-c', 'webpack src/main.js dist/bundle.js' ]
12 silly lifecycle webpack-practice@1.0.0~build: Returned: code: 2  signal: null
13 info lifecycle webpack-practice@1.0.0~build: Failed to exec build script
14 verbose stack Error: webpack-practice@1.0.0 build: `webpack src/main.js dist/bundle.js`
14 verbose stack Exit status 2
14 verbose stack     at EventEmitter.<anonymous> (/usr/share/npm/lib/utils/lifecycle.js:232:16)
14 verbose stack     at emitTwo (events.js:126:13)
14 verbose stack     at EventEmitter.emit (events.js:214:7)
14 verbose stack     at ChildProcess.<anonymous> (/usr/share/npm/lib/utils/spawn.js:24:14)
14 verbose stack     at emitTwo (events.js:126:13)
14 verbose stack     at ChildProcess.emit (events.js:214:7)
14 verbose stack     at maybeClose (internal/child_process.js:925:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
15 verbose pkgid webpack-practice@1.0.0
16 verbose cwd /mnt/c/Users/Owner/Work/Web/Webpack-Practice
17 error Linux 4.4.0-17134-Microsoft
18 error argv "/usr/bin/node" "/usr/bin/npm" "run" "build"
19 error node v8.10.0
20 error npm  v3.5.2
21 error code ELIFECYCLE
22 error webpack-practice@1.0.0 build: `webpack src/main.js dist/bundle.js`
22 error Exit status 2
23 error Failed at the webpack-practice@1.0.0 build script 'webpack src/main.js dist/bundle.js'.
23 error Make sure you have the latest version of node.js and npm installed.
23 error If you do, this is most likely a problem with the webpack-practice package,
23 error not with npm itself.
23 error Tell the author that this fails on your system:
23 error     webpack src/main.js dist/bundle.js
23 error You can get information on how to open an issue for this project with:
23 error     npm bugs webpack-practice
23 error Or if that isn't available, you can get their info via:
23 error     npm owner ls webpack-practice
23 error There is likely additional logging output above.
24 verbose exit [ 1, true ]

```

One of the messages said I should check and see if I have the latest versions of node and npm so I updated both and it's still giving me these issues. 

And clue what I'm doing wrong?

Also I'm on windows, using Bash and Windows Subsystem Layer.

**Comment by jcklpe** [comment]
Okay so checking the error response in the terminal I noticed it said : 

```
ERROR in multi ./src/main.js dist/bundle.js
Module not found: Error: Can't resolve 'dist/bundle.js' in '/mnt/c/Users/Owner/Work/Web/Webpack-Practice'
 @ multi ./src/main.js dist/bundle.js
```

So I took that to mean that it was throwing an error because I didn't have an existing dist or bundle.js. So I created both. And then ran `npm run build` aaaaand

.... it didn't throw an error. But bundle.js is empty and now there's a copy of main.js in the dist folder for some reason???

**Comment by jcklpe** [comment]
I'm wondering if it has something to do with the fact that when I created package.json with npm init it asked me what my entry was, and I just hit enter. So my package.json looks like this:

```
{
  "name": "webpack-practice",
  "version": "1.0.0",
  "description": "project to practice webpack",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "build": "webpack src/main.js dist/bundle.js"
  },
  "author": "David Aslan French",
  "license": "ISC",
  "devDependencies": {
    "webpack": "^4.12.0",
    "webpack-cli": "^3.0.8"
  },
  "dependencies": {
    "lodash": "^4.17.10"
  }
}
```

So as you can see my main is set to index.js... so neither main or bundle.... so maybe that's affecting it?

**Comment by jcklpe** [comment]
Okay so ignoring the weird main.js in the dist folder thing I went ahead and I got to the end of the example 1 section. But I can't run `npm run start` because the del-cli command for dist folder results in no bundle.js, and as I said in the last one, results in an error being thrown. 😞 Not sure what to do. 

I'm going to try seeing if I reverse the slashes if that will help. Maybe its' because of the stupid linux and windows slash differences?

**Comment by jcklpe** [comment]
Hey @joezimjs , I also tried using your example 1 repo package.json file and it ran into the same error. Fallowing some of the discussion [[here]](https://github.com/webpack/webpack/issues/6858) I think it may have something to do with webpack trying to add on the `./` when it's missing and confusing windows but I'm not sure. I'm also going to post on stack overflow and see if anyone can lend some advice. 

Thank you for putting together this tutorial!

**Comment by jcklpe** [comment]
Ooooh kay. One last thing I decided to try. I was closing out all my tabs and I checked the official documentation for webpack and they had a thing about the command line where it was input and output but between them was `-o`. I don't know what that means. But it works now....I'm going to go look that up. 

[[here's a link]](https://webpack.js.org/api/cli/#usage-without-config-file)

EDIT: I could not find anything about the -o   It's also wicked hard to google it.
